{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2b8eba3f-3e11-4358-88a6-741980dfc9d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import rpy2.ipython\n",
    "rpy2.ipython.load_ipython_extension(get_ipython())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "b9d8958e-ed2e-459e-ae54-1390ae81ca45",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Loading required package: SummarizedExperiment\n",
      "\n",
      "R[write to console]: Loading required package: MatrixGenerics\n",
      "\n",
      "R[write to console]: Loading required package: matrixStats\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: GenomicRanges\n",
      "\n",
      "R[write to console]: Loading required package: stats4\n",
      "\n",
      "R[write to console]: Loading required package: BiocGenerics\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: S4Vectors\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: IRanges\n",
      "\n",
      "R[write to console]: Loading required package: GenomeInfoDb\n",
      "\n",
      "R[write to console]: Loading required package: Biobase\n",
      "\n",
      "R[write to console]: Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: limma\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘limma’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: igraph\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘igraph’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    normalize, path, union\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    decompose, spectrum\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:base’:\n",
      "\n",
      "    union\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: AnnotationHub\n",
      "\n",
      "R[write to console]: Loading required package: BiocFileCache\n",
      "\n",
      "R[write to console]: Loading required package: dbplyr\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘AnnotationHub’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    cache\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: MultiAssayExperiment\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# =================== INSTALACIÓN DE LIBRERIAS ============================================\n",
    "# Instalar y cargar HiPathia\n",
    "library(recount3) \n",
    "library(edgeR)\n",
    "library(SummarizedExperiment)\n",
    "library(hipathia) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "da6dbe54-51c9-4e96-b4cf-a9bc64d81719",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 08:30:58.179266 caching file sra.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:30:59.185718 caching file gtex.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:00.634472 caching file tcga.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:04.449474 downloading and reading the metadata.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:05.905392 caching file gtex.gtex.BLOOD.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:07.076121 caching file gtex.recount_project.BLOOD.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:08.214907 caching file gtex.recount_qc.BLOOD.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:09.543756 caching file gtex.recount_seq_qc.BLOOD.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:10.466407 downloading and reading the feature information.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:11.497112 caching file human.gene_sums.G026.gtf.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:11.978213 downloading and reading the counts: 1048 samples across 63856 features.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:13.238724 caching file gtex.gene_sums.BLOOD.G026.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:21.573633 constructing the RangedSummarizedExperiment (rse) object.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:23.088441 caching file sra.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:24.057137 caching file gtex.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:24.817602 caching file tcga.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:27.003161 downloading and reading the metadata.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:27.972005 caching file gtex.gtex.THYROID.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:28.940076 caching file gtex.recount_project.THYROID.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:29.988309 caching file gtex.recount_qc.THYROID.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:31.253263 caching file gtex.recount_seq_qc.THYROID.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:32.270822 downloading and reading the feature information.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:32.910689 caching file human.gene_sums.G026.gtf.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:33.358178 downloading and reading the counts: 706 samples across 63856 features.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:34.421318 caching file gtex.gene_sums.THYROID.G026.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 08:31:37.322413 constructing the RangedSummarizedExperiment (rse) object.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# =================== OBTENCIÓN DE DATOS ==================================================\n",
    "####-- BLOOD\n",
    "gtex_blood <- subset(available_projects(), file_source == \"gtex\" & \n",
    "                           project ==\"BLOOD\")\n",
    "rseBLOOD <- create_rse(gtex_blood)\n",
    "\n",
    "####-- THYROID\n",
    "gtex_thyr <- subset(available_projects(), file_source == \"gtex\" & \n",
    "                          project ==\"THYROID\")\n",
    "rseTHYR<- create_rse(gtex_thyr)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4700be66-2ec5-4f92-ab73-65268406b62e",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:12:20.932965 caching file sra.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:22.990201 caching file gtex.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:24.250041 caching file tcga.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:28.034564 downloading and reading the metadata.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:28.927535 caching file gtex.gtex.OVARY.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:30.050626 caching file gtex.recount_project.OVARY.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:31.749164 caching file gtex.recount_qc.OVARY.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:32.755553 caching file gtex.recount_seq_qc.OVARY.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:33.691563 downloading and reading the feature information.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:34.968573 caching file human.gene_sums.G026.gtf.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:35.486428 downloading and reading the counts: 195 samples across 63856 features.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:36.549861 caching file gtex.gene_sums.OVARY.G026.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:39.087824 constructing the RangedSummarizedExperiment (rse) object.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:39.881418 caching file sra.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:40.60063 caching file gtex.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:41.385329 caching file tcga.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:44.20698 downloading and reading the metadata.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:45.112744 caching file gtex.gtex.LUNG.MD.gz.\n",
      "\n",
      "R[write to console]: adding rname 'http://duffel.rail.bio/recount3/human/data_sources/gtex/metadata/NG/LUNG/gtex.gtex.LUNG.MD.gz'\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 450 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 450 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 450 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |==========                                                            |  14%\n",
      "  |                                                                            \n",
      "  |===================                                                   |  28%\n",
      "  |                                                                            \n",
      "  |==============================                                        |  43%\n",
      "  |                                                                            \n",
      "  |========================================                              |  57%\n",
      "  |                                                                            \n",
      "  |===========================================================           |  85%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:12:49.066569 caching file gtex.recount_project.LUNG.MD.gz.\n",
      "\n",
      "R[write to console]: adding rname 'http://duffel.rail.bio/recount3/human/data_sources/gtex/metadata/NG/LUNG/gtex.recount_project.LUNG.MD.gz'\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:12:52.342577 caching file gtex.recount_qc.LUNG.MD.gz.\n",
      "\n",
      "R[write to console]: adding rname 'http://duffel.rail.bio/recount3/human/data_sources/gtex/metadata/NG/LUNG/gtex.recount_qc.LUNG.MD.gz'\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |=======                                                               |  10%\n",
      "  |                                                                            \n",
      "  |==============                                                        |  20%\n",
      "  |                                                                            \n",
      "  |====================================                                  |  51%\n",
      "  |                                                                            \n",
      "  |===========================================                           |  62%\n",
      "  |                                                                            \n",
      "  |================================================                      |  69%\n",
      "  |                                                                            \n",
      "  |========================================================              |  79%\n",
      "  |                                                                            \n",
      "  |===============================================================       |  90%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:12:55.596136 caching file gtex.recount_seq_qc.LUNG.MD.gz.\n",
      "\n",
      "R[write to console]: adding rname 'http://duffel.rail.bio/recount3/human/data_sources/gtex/metadata/NG/LUNG/gtex.recount_seq_qc.LUNG.MD.gz'\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 470 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |===============================                                       |  44%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:12:58.409551 downloading and reading the feature information.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:12:59.546561 caching file human.gene_sums.G026.gtf.gz.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:13:00.047173 downloading and reading the counts: 655 samples across 63856 features.\n",
      "\n",
      "R[write to console]: 2025-10-25 20:13:00.946337 caching file gtex.gene_sums.LUNG.G026.gz.\n",
      "\n",
      "R[write to console]: adding rname 'http://duffel.rail.bio/recount3/human/data_sources/gtex/gene_sums/NG/LUNG/gtex.gene_sums.LUNG.G026.gz'\n",
      "\n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "Downloading: nsole]: \n",
      "R[write to console]: \n",
      "R[write to console]: 460 B\n",
      "R[write to console]: \n",
      "R[write to console]:      \n",
      "  |                                                                            \n",
      "  |                                                                      |   0%\n",
      "  |                                                                            \n",
      "  |                                                                      |   1%\n",
      "  |                                                                            \n",
      "  |=                                                                     |   1%\n",
      "  |                                                                            \n",
      "  |=                                                                     |   2%\n",
      "  |                                                                            \n",
      "  |==                                                                    |   2%\n",
      "  |                                                                            \n",
      "  |==                                                                    |   3%\n",
      "  |                                                                            \n",
      "  |==                                                                    |   4%\n",
      "  |                                                                            \n",
      "  |===                                                                   |   4%\n",
      "  |                                                                            \n",
      "  |===                                                                   |   5%\n",
      "  |                                                                            \n",
      "  |====                                                                  |   5%\n",
      "  |                                                                            \n",
      "  |====                                                                  |   6%\n",
      "  |                                                                            \n",
      "  |=====                                                                 |   6%\n",
      "  |                                                                            \n",
      "  |=====                                                                 |   7%\n",
      "  |                                                                            \n",
      "  |=====                                                                 |   8%\n",
      "  |                                                                            \n",
      "  |======                                                                |   8%\n",
      "  |                                                                            \n",
      "  |======                                                                |   9%\n",
      "  |                                                                            \n",
      "  |=======                                                               |   9%\n",
      "  |                                                                            \n",
      "  |=======                                                               |  10%\n",
      "  |                                                                            \n",
      "  |=======                                                               |  11%\n",
      "  |                                                                            \n",
      "  |========                                                              |  11%\n",
      "  |                                                                            \n",
      "  |========                                                              |  12%\n",
      "  |                                                                            \n",
      "  |=========                                                             |  12%\n",
      "  |                                                                            \n",
      "  |=========                                                             |  13%\n",
      "  |                                                                            \n",
      "  |=========                                                             |  14%\n",
      "  |                                                                            \n",
      "  |==========                                                            |  14%\n",
      "  |                                                                            \n",
      "  |==========                                                            |  15%\n",
      "  |                                                                            \n",
      "  |===========                                                           |  15%\n",
      "  |                                                                            \n",
      "  |===========                                                           |  16%\n",
      "  |                                                                            \n",
      "  |============                                                          |  16%\n",
      "  |                                                                            \n",
      "  |============                                                          |  17%\n",
      "  |                                                                            \n",
      "  |============                                                          |  18%\n",
      "  |                                                                            \n",
      "  |=============                                                         |  18%\n",
      "  |                                                                            \n",
      "  |=============                                                         |  19%\n",
      "  |                                                                            \n",
      "  |==============                                                        |  19%\n",
      "  |                                                                            \n",
      "  |==============                                                        |  20%\n",
      "  |                                                                            \n",
      "  |==============                                                        |  21%\n",
      "  |                                                                            \n",
      "  |===============                                                       |  21%\n",
      "  |                                                                            \n",
      "  |===============                                                       |  22%\n",
      "  |                                                                            \n",
      "  |================                                                      |  22%\n",
      "  |                                                                            \n",
      "  |================                                                      |  23%\n",
      "  |                                                                            \n",
      "  |================                                                      |  24%\n",
      "  |                                                                            \n",
      "  |=================                                                     |  24%\n",
      "  |                                                                            \n",
      "  |=================                                                     |  25%\n",
      "  |                                                                            \n",
      "  |==================                                                    |  25%\n",
      "  |                                                                            \n",
      "  |==================                                                    |  26%\n",
      "  |                                                                            \n",
      "  |===================                                                   |  26%\n",
      "  |                                                                            \n",
      "  |===================                                                   |  27%\n",
      "  |                                                                            \n",
      "  |===================                                                   |  28%\n",
      "  |                                                                            \n",
      "  |====================                                                  |  28%\n",
      "  |                                                                            \n",
      "  |====================                                                  |  29%\n",
      "  |                                                                            \n",
      "  |=====================                                                 |  29%\n",
      "  |                                                                            \n",
      "  |=====================                                                 |  30%\n",
      "  |                                                                            \n",
      "  |=====================                                                 |  31%\n",
      "  |                                                                            \n",
      "  |======================                                                |  31%\n",
      "  |                                                                            \n",
      "  |======================                                                |  32%\n",
      "  |                                                                            \n",
      "  |=======================                                               |  32%\n",
      "  |                                                                            \n",
      "  |=======================                                               |  33%\n",
      "  |                                                                            \n",
      "  |=======================                                               |  34%\n",
      "  |                                                                            \n",
      "  |========================                                              |  34%\n",
      "  |                                                                            \n",
      "  |========================                                              |  35%\n",
      "  |                                                                            \n",
      "  |=========================                                             |  35%\n",
      "  |                                                                            \n",
      "  |=========================                                             |  36%\n",
      "  |                                                                            \n",
      "  |==========================                                            |  36%\n",
      "  |                                                                            \n",
      "  |==========================                                            |  37%\n",
      "  |                                                                            \n",
      "  |==========================                                            |  38%\n",
      "  |                                                                            \n",
      "  |===========================                                           |  38%\n",
      "  |                                                                            \n",
      "  |===========================                                           |  39%\n",
      "  |                                                                            \n",
      "  |============================                                          |  39%\n",
      "  |                                                                            \n",
      "  |============================                                          |  40%\n",
      "  |                                                                            \n",
      "  |============================                                          |  41%\n",
      "  |                                                                            \n",
      "  |=============================                                         |  41%\n",
      "  |                                                                            \n",
      "  |=============================                                         |  42%\n",
      "  |                                                                            \n",
      "  |==============================                                        |  42%\n",
      "  |                                                                            \n",
      "  |==============================                                        |  43%\n",
      "  |                                                                            \n",
      "  |==============================                                        |  44%\n",
      "  |                                                                            \n",
      "  |===============================                                       |  44%\n",
      "  |                                                                            \n",
      "  |===============================                                       |  45%\n",
      "  |                                                                            \n",
      "  |================================                                      |  45%\n",
      "  |                                                                            \n",
      "  |================================                                      |  46%\n",
      "  |                                                                            \n",
      "  |=================================                                     |  46%\n",
      "  |                                                                            \n",
      "  |=================================                                     |  47%\n",
      "  |                                                                            \n",
      "  |=================================                                     |  48%\n",
      "  |                                                                            \n",
      "  |==================================                                    |  48%\n",
      "  |                                                                            \n",
      "  |==================================                                    |  49%\n",
      "  |                                                                            \n",
      "  |===================================                                   |  49%\n",
      "  |                                                                            \n",
      "  |===================================                                   |  50%\n",
      "  |                                                                            \n",
      "  |===================================                                   |  51%\n",
      "  |                                                                            \n",
      "  |====================================                                  |  51%\n",
      "  |                                                                            \n",
      "  |====================================                                  |  52%\n",
      "  |                                                                            \n",
      "  |=====================================                                 |  52%\n",
      "  |                                                                            \n",
      "  |=====================================                                 |  53%\n",
      "  |                                                                            \n",
      "  |=====================================                                 |  54%\n",
      "  |                                                                            \n",
      "  |======================================                                |  54%\n",
      "  |                                                                            \n",
      "  |======================================                                |  55%\n",
      "  |                                                                            \n",
      "  |=======================================                               |  55%\n",
      "  |                                                                            \n",
      "  |=======================================                               |  56%\n",
      "  |                                                                            \n",
      "  |========================================                              |  56%\n",
      "  |                                                                            \n",
      "  |========================================                              |  57%\n",
      "  |                                                                            \n",
      "  |========================================                              |  58%\n",
      "  |                                                                            \n",
      "  |=========================================                             |  58%\n",
      "  |                                                                            \n",
      "  |=========================================                             |  59%\n",
      "  |                                                                            \n",
      "  |==========================================                            |  59%\n",
      "  |                                                                            \n",
      "  |==========================================                            |  60%\n",
      "  |                                                                            \n",
      "  |==========================================                            |  61%\n",
      "  |                                                                            \n",
      "  |===========================================                           |  61%\n",
      "  |                                                                            \n",
      "  |===========================================                           |  62%\n",
      "  |                                                                            \n",
      "  |============================================                          |  62%\n",
      "  |                                                                            \n",
      "  |============================================                          |  63%\n",
      "  |                                                                            \n",
      "  |============================================                          |  64%\n",
      "  |                                                                            \n",
      "  |=============================================                         |  64%\n",
      "  |                                                                            \n",
      "  |=============================================                         |  65%\n",
      "  |                                                                            \n",
      "  |==============================================                        |  65%\n",
      "  |                                                                            \n",
      "  |==============================================                        |  66%\n",
      "  |                                                                            \n",
      "  |===============================================                       |  66%\n",
      "  |                                                                            \n",
      "  |===============================================                       |  67%\n",
      "  |                                                                            \n",
      "  |===============================================                       |  68%\n",
      "  |                                                                            \n",
      "  |================================================                      |  68%\n",
      "  |                                                                            \n",
      "  |================================================                      |  69%\n",
      "  |                                                                            \n",
      "  |=================================================                     |  69%\n",
      "  |                                                                            \n",
      "  |=================================================                     |  70%\n",
      "  |                                                                            \n",
      "  |=================================================                     |  71%\n",
      "  |                                                                            \n",
      "  |==================================================                    |  71%\n",
      "  |                                                                            \n",
      "  |==================================================                    |  72%\n",
      "  |                                                                            \n",
      "  |===================================================                   |  72%\n",
      "  |                                                                            \n",
      "  |===================================================                   |  73%\n",
      "  |                                                                            \n",
      "  |====================================================                  |  74%\n",
      "  |                                                                            \n",
      "  |====================================================                  |  75%\n",
      "  |                                                                            \n",
      "  |=====================================================                 |  75%\n",
      "  |                                                                            \n",
      "  |=====================================================                 |  76%\n",
      "  |                                                                            \n",
      "  |======================================================                |  76%\n",
      "  |                                                                            \n",
      "  |======================================================                |  77%\n",
      "  |                                                                            \n",
      "  |======================================================                |  78%\n",
      "  |                                                                            \n",
      "  |=======================================================               |  78%\n",
      "  |                                                                            \n",
      "  |=======================================================               |  79%\n",
      "  |                                                                            \n",
      "  |========================================================              |  79%\n",
      "  |                                                                            \n",
      "  |========================================================              |  80%\n",
      "  |                                                                            \n",
      "  |========================================================              |  81%\n",
      "  |                                                                            \n",
      "  |=========================================================             |  81%\n",
      "  |                                                                            \n",
      "  |=========================================================             |  82%\n",
      "  |                                                                            \n",
      "  |==========================================================            |  82%\n",
      "  |                                                                            \n",
      "  |==========================================================            |  83%\n",
      "  |                                                                            \n",
      "  |==========================================================            |  84%\n",
      "  |                                                                            \n",
      "  |===========================================================           |  84%\n",
      "  |                                                                            \n",
      "  |===========================================================           |  85%\n",
      "  |                                                                            \n",
      "  |============================================================          |  85%\n",
      "  |                                                                            \n",
      "  |============================================================          |  86%\n",
      "  |                                                                            \n",
      "  |=============================================================         |  86%\n",
      "  |                                                                            \n",
      "  |=============================================================         |  87%\n",
      "  |                                                                            \n",
      "  |=============================================================         |  88%\n",
      "  |                                                                            \n",
      "  |==============================================================        |  88%\n",
      "  |                                                                            \n",
      "  |==============================================================        |  89%\n",
      "  |                                                                            \n",
      "  |===============================================================       |  89%\n",
      "  |                                                                            \n",
      "  |===============================================================       |  90%\n",
      "  |                                                                            \n",
      "  |===============================================================       |  91%\n",
      "  |                                                                            \n",
      "  |================================================================      |  91%\n",
      "  |                                                                            \n",
      "  |================================================================      |  92%\n",
      "  |                                                                            \n",
      "  |=================================================================     |  92%\n",
      "  |                                                                            \n",
      "  |=================================================================     |  93%\n",
      "  |                                                                            \n",
      "  |==================================================================    |  94%\n",
      "  |                                                                            \n",
      "  |==================================================================    |  95%\n",
      "  |                                                                            \n",
      "  |===================================================================   |  95%\n",
      "  |                                                                            \n",
      "  |===================================================================   |  96%\n",
      "  |                                                                            \n",
      "  |====================================================================  |  96%\n",
      "  |                                                                            \n",
      "  |====================================================================  |  97%\n",
      "  |                                                                            \n",
      "  |====================================================================  |  98%\n",
      "  |                                                                            \n",
      "  |===================================================================== |  98%\n",
      "  |                                                                            \n",
      "  |===================================================================== |  99%\n",
      "  |                                                                            \n",
      "  |======================================================================|  99%\n",
      "  |                                                                            \n",
      "  |======================================================================| 100%\n",
      "R[write to console]: \n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-25 20:19:02.667945 constructing the RangedSummarizedExperiment (rse) object.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "#=================================== OTROS TEJIDOS =================================\n",
    "####-- OVARY\n",
    "gtex_ovary <- subset(available_projects(), file_source == \"gtex\" & \n",
    "                          project ==\"OVARY\")\n",
    "rseOVARY<- create_rse(gtex_ovary)\n",
    "\n",
    "####-- OVARY\n",
    "gtex_lung <- subset(available_projects(), file_source == \"gtex\" & \n",
    "                          project ==\"LUNG\")\n",
    "rseLUNG<- create_rse(gtex_lung)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ef564700-b615-407e-8436-6f84eb675ddc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 2025-10-26 10:43:59.824843 caching file sra.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-26 10:44:00.524177 caching file gtex.recount_project.MD.gz.\n",
      "\n",
      "R[write to console]: 2025-10-26 10:44:01.225805 caching file tcga.recount_project.MD.gz.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "gtex_projects <- subset(available_projects(), file_source == \"gtex\")\n",
    "gtex_projects "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ee36833-b2ad-4c6d-9200-b0c9234db7c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# ============================= Limitación de los datos de interes =======================\n",
    "## Limitación de los datos de interes\n",
    "rseBLOOD <- rseBLOOD[, rseBLOOD$gtex.smtsd == \"Whole Blood\" & rseBLOOD$gtex.smafrze == \"RNASEQ\"]\n",
    "rseTHYR <- rseTHYR[, rseTHYR$gtex.smafrze == \"RNASEQ\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b927331b-6bbd-4fb9-a56b-8b486d57c2e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# ============================= OTROS TEJIDOS - Limitación de los datos de interes =========\n",
    "rseLUNG <- rseLUNG[, rseLUNG$gtex.smafrze == \"RNASEQ\"]\n",
    "rseOVARY <- rseOVARY[, rseOVARY$gtex.smafrze == \"RNASEQ\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "795fbffe-b08b-4610-bc70-3b0d05912863",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# ============================= CONTEOS CRUODOS Y METADATA ================================\n",
    "## Obtención por separado de arrays con los conteos crudos de RNA-seq y los \n",
    "## datos clínicos y experimentales en metadata\n",
    "countsBLOOD <- assay(rseBLOOD, \"raw_counts\")  #Matriz de conteos crudos\n",
    "metadataBLOOD <- colData(rseBLOOD)            #df de datos clínicos y experimentales\n",
    "####-- THYROID\n",
    "countsTHYR <- assay(rseTHYR, \"raw_counts\")  #Matriz de conteos crudos\n",
    "metadataTHYR <- colData(rseTHYR)            #df de datos clínicos y experimentales"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "32a05d71-3b8d-43e8-9e0c-a9e0d5da3176",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "#=================================== OTROS TEJIDOS =================================\n",
    "countsLUNG <- assay(rseLUNG, \"raw_counts\")  #Matriz de conteos crudos\n",
    "metadataLUNG<- colData(rseLUNG)            #df de datos clínicos y experimentales\n",
    "####-- THYROID\n",
    "countsOVARY <- assay(rseOVARY, \"raw_counts\")  #Matriz de conteos crudos\n",
    "metadataOVARY <- colData(rseOVARY)            #df de datos clínicos y experimentales"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "18a2f06b-a184-46cd-8bb1-dfa58f3eedc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### ========================  RE-FILTRADO  =================================\n",
    "keepBLOOD <- rowSums(countsBLOOD > 0) >= 5  # Genes expresados en ≥5 muestras\n",
    "countsBLOOD <- countsBLOOD[keepBLOOD, ] #De 63856 genes a 55424 genes\n",
    "####-- THYROID\n",
    "keepTHYR <- rowSums(countsTHYR > 0) >= 5  # Genes expresados en ≥5 muestras\n",
    "countsTHYR <- countsTHYR[keepTHYR, ] #De 63856 genes a 55424 genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "61a7eef7-4097-4bf9-8e26-d9ec7d189027",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### ======================== OTROS TEJIDOS - PRE-FILTRADO ===============================\n",
    "keepLUNG <- rowSums(countsLUNG > 0) >= 5  # Genes expresados en ≥5 muestras\n",
    "countsLUNG <- countsLUNG[keepLUNG, ] #De 63856 genes a 55424 genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e2b919ab-973b-4e7e-95b6-73afda091cc8",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "## ============================ NORMALIZACIÓN TMM EN DATOS CRUDOS =====================\n",
    "    # Normalización TMM = Trimmed Mean of M-values:\n",
    "    # método para corregir diferencias en la profundidad de secuenciación y \n",
    "    # en la composición global del RNA entre muestras.\n",
    "    # Conversión counts crudos a CPM (Counts Per Million),\n",
    "    # usando tamaños de librería ajustados con factores de normalización TMM \n",
    "    # (calcula lib.size * norm.factors)\n",
    "\n",
    "dgeB <- DGEList(counts = countsBLOOD) # clase DGEList (Differential Gene Expression List)\n",
    "dge_normB <- cpm(dgeB, \n",
    "                    normalized.lib.sizes = TRUE, \n",
    "                    log = FALSE)                              \n",
    "                     \n",
    "dge_normalizedB <- log2(dge_normB + 1) # transformación log2 para obtener valores continuos comparables \n",
    "\n",
    "####-- THYROID\n",
    "dgeT <- DGEList(counts = countsTHYR) # clase DGEList (Differential Gene Expression List)\n",
    "dge_normT <- cpm(dgeT, \n",
    "                    normalized.lib.sizes = TRUE, \n",
    "                    log = FALSE)                              \n",
    "                     \n",
    "dge_normalizedT <- log2(dge_normT + 1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "887d7a95-38ba-4301-9b01-650bac549d54",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "## ===================== OTROS TEJIDOS - NORMALIZACIÓN TMM EN DATOS CRUDOS ==================\n",
    "####-- LUNG\n",
    "dgeL <- DGEList(counts = countsLUNG) # clase DGEList (Differential Gene Expression List)\n",
    "dge_normL <- cpm(dgeL, \n",
    "                    normalized.lib.sizes = TRUE, \n",
    "                    log = FALSE)                              \n",
    "                     \n",
    "dge_normalizedL <- log2(dge_normL + 1) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d656003a-602f-49f2-9d19-45517a46ba09",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2023-10-23\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "translated ids = 23275 (0.45) \n",
      "untranslated ids = 27965 (0.55) \n",
      "multihit ids = 172 (0.0034) \n",
      "\n",
      "FALSE \n",
      "23182 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "### ===================== CAMBIO DE ENSEMBL A ENTREZ (Hipathia) ===========================\n",
    "data_entrezB <- translate_data(dge_normalizedB, \"hsa\")\n",
    "      # translated ids = 23736 (0.43) \n",
    "      # untranslated ids = 31688 (0.57) \n",
    "      # multihit ids = 181 (0.0033) \n",
    "    ## De los 55424 genes por Ensembl, se reconocen 23638 con id Entrez en Hipathia\n",
    "table(is.na(rownames(data_entrezB)))\n",
    "    # IMPORTANTE: que no existan filas con id no reconocidos por Hipathia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1ce0822e-2738-446d-b72b-95ca27c83231",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2023-10-23\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "translated ids = 23566 (0.44) \n",
      "untranslated ids = 30445 (0.56) \n",
      "multihit ids = 182 (0.0034) \n",
      "\n",
      "FALSE \n",
      "23468 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "### =========================== CAMBIO DE ENSEMBL A ENTREZ (Hipathia) =====================\n",
    "data_entrezT <- translate_data(dge_normalizedT, \"hsa\")\n",
    "      # translated ids = 23736 (0.43) \n",
    "      # untranslated ids = 31688 (0.57) \n",
    "      # multihit ids = 181 (0.0033) \n",
    "    ## De los 55424 genes por Ensembl, se reconocen 23638 con id Entrez en Hipathia\n",
    "table(is.na(rownames(data_entrezT)))\n",
    "    # IMPORTANTE: que no existan filas con id no reconocidos por Hipathia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4f4de756-dfed-482a-912f-2df94010f629",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2023-10-23\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "translated ids = 23549 (0.44) \n",
      "untranslated ids = 30264 (0.56) \n",
      "multihit ids = 175 (0.0033) \n",
      "\n",
      "FALSE \n",
      "23455 \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "### =========================== CAMBIO DE ENSEMBL A ENTREZ (Hipathia) =====================\n",
    "data_entrezL <- translate_data(dge_normalizedL, \"hsa\")\n",
    "      # translated ids = 23736 (0.43) \n",
    "      # untranslated ids = 31688 (0.57) \n",
    "      # multihit ids = 181 (0.0033) \n",
    "    ## De los 55424 genes por Ensembl, se reconocen 23638 con id Entrez en Hipathia\n",
    "table(is.na(rownames(data_entrezL)))\n",
    "    # IMPORTANTE: que no existan filas con id no reconocidos por Hipathia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "3365e674-7132-4742-a9db-489432505af1",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### ================================ NORMALIZACION a [0,1] (Hipathia) ======================\n",
    "data_entrez_normB <- normalize_data(data_entrezB)\n",
    "data_entrez_normT <- normalize_data(data_entrezT)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "c205b6b0-f3d6-49f0-9e33-84ac626f9d6d",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### ================================ NORMALIZACION a [0,1] (Hipathia) ======================\n",
    "data_entrez_normL <- normalize_data(data_entrezL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "36103c30-6192-4c3e-8aa6-88915b102194",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "UsageError: Cell magic `%%R` not found.\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "# ============== Inversión de columnas por filas ==========================================\n",
    "data_entrez_normB_t <- t(data_entrez_normB)\n",
    "data_entrez_normB_t[1:5, 1:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a3b1c2cf-b8e8-425c-9e6c-9407548537b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# Creación de carpetas\n",
    "if (!dir.exists(\"data\")) {\n",
    "  dir.create(\"data\")\n",
    "}\n",
    "if (!dir.exists(\"data/imagen\")) {\n",
    "  dir.create(\"data/imagen\")\n",
    "}\n",
    "\n",
    "# Guardar datos listos de genes BLOOD como CSV en la carpeta datos\n",
    "write.csv(data_entrez_normB_t, file = \"data/RNAseq_BLOOD.csv\", row.names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d25f3515-8230-4e05-a4ca-4f388b758e86",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### ======================= OBJETO FINAL PARA HIPATHIA SummarizedEperiment ======================\n",
    "data_infoT <- SummarizedExperiment(\n",
    "    assays = list(counts = data_entrez_normT), #counts normalizados\n",
    "     colData = DataFrame(\n",
    "        tissue = colData(rseTHYR)$gtex.smts,\n",
    "        subtissue = colData(rseTHYR)$gtex.smtsd\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "42815ba7-1819-48e7-8d35-bc4057c17488",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "### 6 - OBJETO FINAL PARA HIPATHIA SummarizedEperiment\n",
    "data_infoL <- SummarizedExperiment(\n",
    "    assays = list(counts = data_entrez_normL), #counts normalizados\n",
    "     colData = DataFrame(\n",
    "        tissue = colData(rseLUNG)$gtex.smts,\n",
    "        subtissue = colData(rseLUNG)$gtex.smtsd\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "779b40a7-6d40-4ff5-851b-da60052234ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: snapshotDate(): 2023-10-23\n",
      "\n",
      "R[write to console]: Loaded 146 pathways\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "### ================== DESCARGA PATHWAYS HUMANOS (Hipathia) =================================\n",
    "pathways <- load_pathways(\"hsa\")\n",
    "        # Loaded 146 pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "e1b88f12-5160-4365-bd89-07f741c810bb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Added missing genes: 90 (0.38%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Computing pathway...\n",
      "hsa03320 - PPAR signaling pathway \n",
      "hsa03460 - Fanconi anemia pathway \n",
      "hsa04010 - MAPK signaling pathway \n",
      "hsa04012 - ErbB signaling pathway \n",
      "hsa04014 - Ras signaling pathway \n",
      "hsa04015 - Rap1 signaling pathway \n",
      "hsa04020 - Calcium signaling pathway \n",
      "hsa04022 - cGMP-PKG signaling pathway \n",
      "hsa04024 - cAMP signaling pathway \n",
      "hsa04062 - Chemokine signaling pathway \n",
      "hsa04064 - NF-kappa B signaling pathway \n",
      "hsa04066 - HIF-1 signaling pathway \n",
      "hsa04068 - FoxO signaling pathway \n",
      "hsa04071 - Sphingolipid signaling pathway \n",
      "hsa04072 - Phospholipase D signaling pathway \n",
      "hsa04110 - Cell cycle \n",
      "hsa04114 - Oocyte meiosis \n",
      "hsa04115 - p53 signaling pathway \n",
      "hsa04150 - mTOR signaling pathway \n",
      "hsa04151 - PI3K-Akt signaling pathway \n",
      "hsa04152 - AMPK signaling pathway \n",
      "hsa04210 - Apoptosis \n",
      "hsa04211 - Longevity regulating pathway - mammal \n",
      "hsa04213 - Longevity regulating pathway - multiple species \n",
      "hsa04218 - Cellular senescence \n",
      "hsa04261 - Adrenergic signaling in cardiomyocytes \n",
      "hsa04270 - Vascular smooth muscle contraction \n",
      "hsa04310 - Wnt signaling pathway \n",
      "hsa04330 - Notch signaling pathway \n",
      "hsa04340 - Hedgehog signaling pathway \n",
      "hsa04350 - TGF-beta signaling pathway \n",
      "hsa04360 - Axon guidance \n",
      "hsa04370 - VEGF signaling pathway \n",
      "hsa04380 - Osteoclast differentiation \n",
      "hsa04390 - Hippo signaling pathway \n",
      "hsa04510 - Focal adhesion \n",
      "hsa04520 - Adherens junction \n",
      "hsa04530 - Tight junction \n",
      "hsa04540 - Gap junction \n",
      "hsa04550 - Signaling pathways regulating pluripotency of stem cells \n",
      "hsa04610 - Complement and coagulation cascades \n",
      "hsa04611 - Platelet activation \n",
      "hsa04612 - Antigen processing and presentation \n",
      "hsa04620 - Toll-like receptor signaling pathway \n",
      "hsa04621 - NOD-like receptor signaling pathway \n",
      "hsa04622 - RIG-I-like receptor signaling pathway \n",
      "hsa04623 - Cytosolic DNA-sensing pathway \n",
      "hsa04630 - Jak-STAT signaling pathway \n",
      "hsa04650 - Natural killer cell mediated cytotoxicity \n",
      "hsa04660 - T cell receptor signaling pathway \n",
      "hsa04662 - B cell receptor signaling pathway \n",
      "hsa04664 - Fc epsilon RI signaling pathway \n",
      "hsa04666 - Fc gamma R-mediated phagocytosis \n",
      "hsa04668 - TNF signaling pathway \n",
      "hsa04670 - Leukocyte transendothelial migration \n",
      "hsa04710 - Circadian rhythm \n",
      "hsa04713 - Circadian entrainment \n",
      "hsa04720 - Long-term potentiation \n",
      "hsa04722 - Neurotrophin signaling pathway \n",
      "hsa04723 - Retrograde endocannabinoid signaling \n",
      "hsa04724 - Glutamatergic synapse \n",
      "hsa04725 - Cholinergic synapse \n",
      "hsa04726 - Serotonergic synapse \n",
      "hsa04727 - GABAergic synapse \n",
      "hsa04728 - Dopaminergic synapse \n",
      "hsa04730 - Long-term depression \n",
      "hsa04740 - Olfactory transduction \n",
      "hsa04742 - Taste transduction \n",
      "hsa04750 - Inflammatory mediator regulation of TRP channels \n",
      "hsa04810 - Regulation of actin cytoskeleton \n",
      "hsa04910 - Insulin signaling pathway \n",
      "hsa04911 - Insulin secretion \n",
      "hsa04912 - GnRH signaling pathway \n",
      "hsa04913 - Ovarian steroidogenesis \n",
      "hsa04914 - Progesterone-mediated oocyte maturation \n",
      "hsa04915 - Estrogen signaling pathway \n",
      "hsa04916 - Melanogenesis \n",
      "hsa04917 - Prolactin signaling pathway \n",
      "hsa04918 - Thyroid hormone synthesis \n",
      "hsa04919 - Thyroid hormone signaling pathway \n",
      "hsa04920 - Adipocytokine signaling pathway \n",
      "hsa04921 - Oxytocin signaling pathway \n",
      "hsa04922 - Glucagon signaling pathway \n",
      "hsa04923 - Regulation of lipolysis in adipocytes \n",
      "hsa04924 - Renin secretion \n",
      "hsa04925 - Aldosterone synthesis and secretion \n",
      "hsa04930 - Type II diabetes mellitus \n",
      "hsa04931 - Insulin resistance \n",
      "hsa04932 - Non-alcoholic fatty liver disease (NAFLD) \n",
      "hsa04933 - AGE-RAGE signaling pathway in diabetic complications \n",
      "hsa04950 - Maturity onset diabetes of the young \n",
      "hsa04960 - Aldosterone-regulated sodium reabsorption \n",
      "hsa04961 - Endocrine and other factor-regulated calcium reabsorption \n",
      "hsa04962 - Vasopressin-regulated water reabsorption \n",
      "hsa04970 - Salivary secretion \n",
      "hsa04971 - Gastric acid secretion \n",
      "hsa04972 - Pancreatic secretion \n",
      "hsa04973 - Carbohydrate digestion and absorption \n",
      "hsa04976 - Bile secretion \n",
      "hsa05010 - Alzheimer's disease \n",
      "hsa05012 - Parkinson's disease \n",
      "hsa05014 - Amyotrophic lateral sclerosis (ALS) \n",
      "hsa05016 - Huntington's disease \n",
      "hsa05020 - Prion diseases \n",
      "hsa05030 - Cocaine addiction \n",
      "hsa05031 - Amphetamine addiction \n",
      "hsa05032 - Morphine addiction \n",
      "hsa05034 - Alcoholism \n",
      "hsa05100 - Bacterial invasion of epithelial cells \n",
      "hsa05110 - Vibrio cholerae infection \n",
      "hsa05120 - Epithelial cell signaling in Helicobacter pylori infection \n",
      "hsa05130 - Pathogenic Escherichia coli infection \n",
      "hsa05131 - Shigellosis \n",
      "hsa05132 - Salmonella infection \n",
      "hsa05133 - Pertussis \n",
      "hsa05134 - Legionellosis \n",
      "hsa05140 - Leishmaniasis \n",
      "hsa05142 - Chagas disease (American trypanosomiasis) \n",
      "hsa05145 - Toxoplasmosis \n",
      "hsa05150 - Staphylococcus aureus infection \n",
      "hsa05152 - Tuberculosis \n",
      "hsa05160 - Hepatitis C \n",
      "hsa05161 - Hepatitis B \n",
      "hsa05162 - Measles \n",
      "hsa05164 - Influenza A \n",
      "hsa05166 - HTLV-I infection \n",
      "hsa05168 - Herpes simplex infection \n",
      "hsa05169 - Epstein-Barr virus infection \n",
      "hsa05200 - Pathways in cancer \n",
      "hsa05205 - Proteoglycans in cancer \n",
      "hsa05210 - Colorectal cancer \n",
      "hsa05211 - Renal cell carcinoma \n",
      "hsa05212 - Pancreatic cancer \n",
      "hsa05213 - Endometrial cancer \n",
      "hsa05214 - Glioma \n",
      "hsa05215 - Prostate cancer \n",
      "hsa05216 - Thyroid cancer \n",
      "hsa05217 - Basal cell carcinoma \n",
      "hsa05218 - Melanoma \n",
      "hsa05219 - Bladder cancer \n",
      "hsa05220 - Chronic myeloid leukemia \n",
      "hsa05221 - Acute myeloid leukemia \n",
      "hsa05222 - Small cell lung cancer \n",
      "hsa05223 - Non-small cell lung cancer \n",
      "hsa05231 - Choline metabolism in cancer \n",
      "hsa05321 - Inflammatory bowel disease (IBD) \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "    \n",
    "### ======================= CALCULO DE ACTIVIDAD DE RUTAS =========================================\n",
    "## HIPATHIA - result\n",
    "results <- hipathia(data_infoT, pathways) # utiliza las subrutas efectoras\n",
    "                                             # opción decompose=TRUE utiliza las subrutas descompuestas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "dd522980-5273-450b-8b23-b644e684d722",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: Added missing genes: 81 (0.34%)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Computing pathway...\n",
      "hsa03320 - PPAR signaling pathway \n",
      "hsa03460 - Fanconi anemia pathway \n",
      "hsa04010 - MAPK signaling pathway \n",
      "hsa04012 - ErbB signaling pathway \n",
      "hsa04014 - Ras signaling pathway \n",
      "hsa04015 - Rap1 signaling pathway \n",
      "hsa04020 - Calcium signaling pathway \n",
      "hsa04022 - cGMP-PKG signaling pathway \n",
      "hsa04024 - cAMP signaling pathway \n",
      "hsa04062 - Chemokine signaling pathway \n",
      "hsa04064 - NF-kappa B signaling pathway \n",
      "hsa04066 - HIF-1 signaling pathway \n",
      "hsa04068 - FoxO signaling pathway \n",
      "hsa04071 - Sphingolipid signaling pathway \n",
      "hsa04072 - Phospholipase D signaling pathway \n",
      "hsa04110 - Cell cycle \n",
      "hsa04114 - Oocyte meiosis \n",
      "hsa04115 - p53 signaling pathway \n",
      "hsa04150 - mTOR signaling pathway \n",
      "hsa04151 - PI3K-Akt signaling pathway \n",
      "hsa04152 - AMPK signaling pathway \n",
      "hsa04210 - Apoptosis \n",
      "hsa04211 - Longevity regulating pathway - mammal \n",
      "hsa04213 - Longevity regulating pathway - multiple species \n",
      "hsa04218 - Cellular senescence \n",
      "hsa04261 - Adrenergic signaling in cardiomyocytes \n",
      "hsa04270 - Vascular smooth muscle contraction \n",
      "hsa04310 - Wnt signaling pathway \n",
      "hsa04330 - Notch signaling pathway \n",
      "hsa04340 - Hedgehog signaling pathway \n",
      "hsa04350 - TGF-beta signaling pathway \n",
      "hsa04360 - Axon guidance \n",
      "hsa04370 - VEGF signaling pathway \n",
      "hsa04380 - Osteoclast differentiation \n",
      "hsa04390 - Hippo signaling pathway \n",
      "hsa04510 - Focal adhesion \n",
      "hsa04520 - Adherens junction \n",
      "hsa04530 - Tight junction \n",
      "hsa04540 - Gap junction \n",
      "hsa04550 - Signaling pathways regulating pluripotency of stem cells \n",
      "hsa04610 - Complement and coagulation cascades \n",
      "hsa04611 - Platelet activation \n",
      "hsa04612 - Antigen processing and presentation \n",
      "hsa04620 - Toll-like receptor signaling pathway \n",
      "hsa04621 - NOD-like receptor signaling pathway \n",
      "hsa04622 - RIG-I-like receptor signaling pathway \n",
      "hsa04623 - Cytosolic DNA-sensing pathway \n",
      "hsa04630 - Jak-STAT signaling pathway \n",
      "hsa04650 - Natural killer cell mediated cytotoxicity \n",
      "hsa04660 - T cell receptor signaling pathway \n",
      "hsa04662 - B cell receptor signaling pathway \n",
      "hsa04664 - Fc epsilon RI signaling pathway \n",
      "hsa04666 - Fc gamma R-mediated phagocytosis \n",
      "hsa04668 - TNF signaling pathway \n",
      "hsa04670 - Leukocyte transendothelial migration \n",
      "hsa04710 - Circadian rhythm \n",
      "hsa04713 - Circadian entrainment \n",
      "hsa04720 - Long-term potentiation \n",
      "hsa04722 - Neurotrophin signaling pathway \n",
      "hsa04723 - Retrograde endocannabinoid signaling \n",
      "hsa04724 - Glutamatergic synapse \n",
      "hsa04725 - Cholinergic synapse \n",
      "hsa04726 - Serotonergic synapse \n",
      "hsa04727 - GABAergic synapse \n",
      "hsa04728 - Dopaminergic synapse \n",
      "hsa04730 - Long-term depression \n",
      "hsa04740 - Olfactory transduction \n",
      "hsa04742 - Taste transduction \n",
      "hsa04750 - Inflammatory mediator regulation of TRP channels \n",
      "hsa04810 - Regulation of actin cytoskeleton \n",
      "hsa04910 - Insulin signaling pathway \n",
      "hsa04911 - Insulin secretion \n",
      "hsa04912 - GnRH signaling pathway \n",
      "hsa04913 - Ovarian steroidogenesis \n",
      "hsa04914 - Progesterone-mediated oocyte maturation \n",
      "hsa04915 - Estrogen signaling pathway \n",
      "hsa04916 - Melanogenesis \n",
      "hsa04917 - Prolactin signaling pathway \n",
      "hsa04918 - Thyroid hormone synthesis \n",
      "hsa04919 - Thyroid hormone signaling pathway \n",
      "hsa04920 - Adipocytokine signaling pathway \n",
      "hsa04921 - Oxytocin signaling pathway \n",
      "hsa04922 - Glucagon signaling pathway \n",
      "hsa04923 - Regulation of lipolysis in adipocytes \n",
      "hsa04924 - Renin secretion \n",
      "hsa04925 - Aldosterone synthesis and secretion \n",
      "hsa04930 - Type II diabetes mellitus \n",
      "hsa04931 - Insulin resistance \n",
      "hsa04932 - Non-alcoholic fatty liver disease (NAFLD) \n",
      "hsa04933 - AGE-RAGE signaling pathway in diabetic complications \n",
      "hsa04950 - Maturity onset diabetes of the young \n",
      "hsa04960 - Aldosterone-regulated sodium reabsorption \n",
      "hsa04961 - Endocrine and other factor-regulated calcium reabsorption \n",
      "hsa04962 - Vasopressin-regulated water reabsorption \n",
      "hsa04970 - Salivary secretion \n",
      "hsa04971 - Gastric acid secretion \n",
      "hsa04972 - Pancreatic secretion \n",
      "hsa04973 - Carbohydrate digestion and absorption \n",
      "hsa04976 - Bile secretion \n",
      "hsa05010 - Alzheimer's disease \n",
      "hsa05012 - Parkinson's disease \n",
      "hsa05014 - Amyotrophic lateral sclerosis (ALS) \n",
      "hsa05016 - Huntington's disease \n",
      "hsa05020 - Prion diseases \n",
      "hsa05030 - Cocaine addiction \n",
      "hsa05031 - Amphetamine addiction \n",
      "hsa05032 - Morphine addiction \n",
      "hsa05034 - Alcoholism \n",
      "hsa05100 - Bacterial invasion of epithelial cells \n",
      "hsa05110 - Vibrio cholerae infection \n",
      "hsa05120 - Epithelial cell signaling in Helicobacter pylori infection \n",
      "hsa05130 - Pathogenic Escherichia coli infection \n",
      "hsa05131 - Shigellosis \n",
      "hsa05132 - Salmonella infection \n",
      "hsa05133 - Pertussis \n",
      "hsa05134 - Legionellosis \n",
      "hsa05140 - Leishmaniasis \n",
      "hsa05142 - Chagas disease (American trypanosomiasis) \n",
      "hsa05145 - Toxoplasmosis \n",
      "hsa05150 - Staphylococcus aureus infection \n",
      "hsa05152 - Tuberculosis \n",
      "hsa05160 - Hepatitis C \n",
      "hsa05161 - Hepatitis B \n",
      "hsa05162 - Measles \n",
      "hsa05164 - Influenza A \n",
      "hsa05166 - HTLV-I infection \n",
      "hsa05168 - Herpes simplex infection \n",
      "hsa05169 - Epstein-Barr virus infection \n",
      "hsa05200 - Pathways in cancer \n",
      "hsa05205 - Proteoglycans in cancer \n",
      "hsa05210 - Colorectal cancer \n",
      "hsa05211 - Renal cell carcinoma \n",
      "hsa05212 - Pancreatic cancer \n",
      "hsa05213 - Endometrial cancer \n",
      "hsa05214 - Glioma \n",
      "hsa05215 - Prostate cancer \n",
      "hsa05216 - Thyroid cancer \n",
      "hsa05217 - Basal cell carcinoma \n",
      "hsa05218 - Melanoma \n",
      "hsa05219 - Bladder cancer \n",
      "hsa05220 - Chronic myeloid leukemia \n",
      "hsa05221 - Acute myeloid leukemia \n",
      "hsa05222 - Small cell lung cancer \n",
      "hsa05223 - Non-small cell lung cancer \n",
      "hsa05231 - Choline metabolism in cancer \n",
      "hsa05321 - Inflammatory bowel disease (IBD) \n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "### ================ OTRO TEJIDO - CALCULO DE ACTIVIDAD DE RUTAS ===================\n",
    "## HIPATHIA - result\n",
    "result <- hipathia(data_infoL, pathways) # utiliza las subrutas efectoras\n",
    "                                             # opción decompose=TRUE utiliza las subrutas descompuestas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d89494a3-9a9a-465e-bb0b-4cd8171a150b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "## ============== Obtención de los valores de actividad de la señal como matriz ===============\n",
    "path_vals <- get_paths_data(results, matrix = TRUE)\n",
    "    ## Obtengo los valores de actividad de la señal como SummarizedExperiment\n",
    "    #path <- get_paths_data(results)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "14987dea-0f7b-4c19-b292-1eeb7f3b1084",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                           P-hsa03320-37 P-hsa03320-61 P-hsa03320-46\n",
      "GTEX-T5JW-1226-SM-3GACY.1    0.010342762    0.01556156  0.0007070438\n",
      "GTEX-WXYG-0226-SM-3NB2Y.1    0.005047896    0.01674471  0.0015471193\n",
      "GTEX-WHPG-0226-SM-3NMB9.1    0.001097455    0.03199625  0.0020552867\n",
      "GTEX-16GPK-0926-SM-6LPJ9.1   0.007428752    0.04668860  0.0038370141\n",
      "GTEX-183WM-2626-SM-7KFRY.1   0.035235921    0.03476212  0.0013578147\n",
      "                           P-hsa03320-57 P-hsa03320-64\n",
      "GTEX-T5JW-1226-SM-3GACY.1    0.001623404  0.0000000000\n",
      "GTEX-WXYG-0226-SM-3NB2Y.1    0.003558841  0.0000000000\n",
      "GTEX-WHPG-0226-SM-3NMB9.1    0.000000000  0.0005287527\n",
      "GTEX-16GPK-0926-SM-6LPJ9.1   0.011437219  0.0035653286\n",
      "GTEX-183WM-2626-SM-7KFRY.1   0.001803365  0.0017746546\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "## =========== Transposición de filas x columna =====================================\n",
    "hipathia_thyr <- t(path_vals)\n",
    "print(hipathia_thyr[0:5,0:5]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "185e1216-909a-48fb-89c1-e328143b7843",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                             P-hsa03320-37 P-hsa03320-61 P-hsa03320-46\n",
      "TCGA.BH.A1FM.11B.23R.A13Q.07     0.3971915     0.2078705     0.1250563\n",
      "TCGA.E2.A1LB.11A.22R.A144.07     0.3558425     0.2107836     0.1234348\n",
      "TCGA.BH.A208.11A.51R.A157.07     0.3757608     0.1847616     0.1242326\n",
      "TCGA.BH.A18K.11A.13R.A12D.07     0.3851692     0.2032974     0.1269767\n",
      "TCGA.E9.A1RC.11A.33R.A157.07     0.1763101     0.2238946     0.1298247\n",
      "                             P-hsa03320-57 P-hsa03320-64\n",
      "TCGA.BH.A1FM.11B.23R.A13Q.07     0.1348015     0.1229489\n",
      "TCGA.E2.A1LB.11A.22R.A144.07     0.1330537     0.1500581\n",
      "TCGA.BH.A208.11A.51R.A157.07     0.1339136     0.1518118\n",
      "TCGA.BH.A18K.11A.13R.A12D.07     0.1645470     0.1483787\n",
      "TCGA.E9.A1RC.11A.33R.A157.07     0.1399415     0.1276370\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "hipathia_LUNG <- t(path_vals)\n",
    "print(hipathia_LUNG[0:5,0:5]) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "07500d9c-8d1d-4ff5-bb6a-29b201f3a24f",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "# Guardar datos listos de rutas Thyroid\n",
    "write.csv(hipathia_LUNG , file = \"data/hipathia_LUNG.csv\", row.names = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "295d0e0c-3ef4-43a4-82f1-fd885dc905a4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (UPO_TFM)",
   "language": "python",
   "name": "upo_tfm"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
